var/spack/repos/builtin/packages/hpx5/package.py

```diff
@@ -39,40 +39,95 @@ class Hpx5(Package):
     homepage = "http://hpx.crest.iu.edu"
     url      = "http://hpx.crest.iu.edu/release/hpx-3.1.0.tar.gz"
 
+    version('4.0.0', 'b40dc03449ae1039cbb48ee149952b22')
     version('3.1.0', '9e90b8ac46788c009079632828c77628')
     version('2.0.0', '3d2ff3aab6c46481f9ec65c5b2bfe7a6')
     version('1.3.0', '2260ecc7f850e71a4d365a43017d8cee')
     version('1.2.0', '4972005f85566af4afe8b71afbf1480f')
     version('1.1.0', '646afb460ecb7e0eea713a634933ce4f')
     version('1.0.0', '8020822adf6090bd59ed7fe465f6c6cb')
 
+    variant('cuda', default=False, description='Enable CUDA support')
+    variant('cxx11', default=False, description='Enable C++11 hpx++ interface')
     variant('debug', default=False, description='Build debug version of HPX-5')
-    variant('photon', default=False, description='Enable Photon support')
+    variant('metis', default=False, description='Enable METIS support')
     variant('mpi', default=False, description='Enable MPI support')
+    variant('opencl', default=False, description='Enable OpenCL support')
+    variant('photon', default=False, description='Enable Photon support')
+    variant('pic', default=True, description='Produce position-independent code')
 
+    depends_on("autoconf", type='build')
+    depends_on("automake", type='build')
+    depends_on("cuda", when='+cuda')
+    depends_on("hwloc")
+    depends_on("jemalloc")
+    depends_on("libffi")
+    depends_on("libtool", type='build')
+    # depends_on("lz4")   # hpx5 always builds its own lz4
+    depends_on("metis", when='+metis')
     depends_on("mpi", when='+mpi')
     depends_on("mpi", when='+photon')
+    depends_on("opencl", when='+opencl')
+    depends_on("pkg-config", type='build')
+
+    configure_directory = "hpx"
+    build_directory = "spack-build"
+
+    def configure_args(self):
+        spec = self.spec
+        args = [
+            '--enable-agas',          # make this a variant?
+            '--enable-jemalloc',      # make this a variant?
+            '--enable-percolation',   # make this a variant?
+            # '--enable-rebalancing',   # this seems broken
+            '--with-hwloc=hwloc',
+            '--with-jemalloc=jemalloc',
+            '--with-libffi=libffi',
+        ]
+
+        if '+cuda' in spec:
+            args += ['--enable-cuda']
+        else:
+            args += ['--disable-cuda']
+
+        if '+cxx11' in spec:
+            args += ['--enable-hpx++']
 
-    def install(self, spec, prefix):
-        extra_args = []
         if '+debug' in spec:
-            extra_args.extend([
-                '--enable-debug',
-                'CFLAGS=-g -O0'
-            ])
+            args += ['--enable-debug']
```

Comment:
I guess that's fine then. I generally try to be explicit in packages. A lot of packages default to "true if the dependencies are on your system, else false". For Spack, this means that `spack install foo ~bar` might install `foo` with `bar` support or it might not, depending on what is on your system. This is obviously not the desired behavior. Adding `--disable-foo` or `--without-foo` solves this problem. But if the default is "false" and not "depends on what is on your system" then that works too.